### PR TITLE
[build] update build scripts and add multi-radio (TREL) CI build checks

### PR DESCRIPTION
**Pull request by abtink**
This commit updates `toranj/build.sh` script for posix based builds to
generate CLI executable. It also updates the posix platform to provide
log functionality when `LOG_OUTPUT` is set to `LOG_OUTPUT_APP`.

This commit also adds CI build checks for multi-radio and TREL build
configurations under posix platform.

-----

This is related to https://github.com/openthread/openthread/issues/6472.

### CODE REVIEW

File: src/posix/platform/logging.cpp
Comment (simonlingoogle) [inline src/posix/platform/logging.cpp:38]: Why define `otPlatLog` in platform when `LOG_OUTPUT == APP` ?

**Review by simonlingoogle — APPROVED** [review]

File: src/posix/platform/logging.cpp
Comment (abtink) [reply src/posix/platform/logging.cpp:38]: Under `posix` build the posix platform itself is acting as the app. If we don't add it the under `LOG_OUTPUT_APP` we would get an error (missing definition of `otPlatLog`). I guess such a config may have not ben covered by any CI build before.

**Comment by abtink** [comment]
@jwhui just a quick note on this. This have now two commits (which I guess we can squash when merging). Thanks.

File: src/posix/platform/logging.cpp
Comment (simonlingoogle) [reply src/posix/platform/logging.cpp:38]: > Under `posix` build the posix platform itself is acting as the app

I think this is not totally true. Should we keep the boundary of posix platform and posix app more clear?

> If we don't add it the under LOG_OUTPUT_APP we would get an error (missing definition of otPlatLog). 

We should fix in a more suitable way?

Thoughts? @bukepo

File: src/posix/platform/trel_udp6.cpp
Comment (bukepo) [inline src/posix/platform/trel_udp6.cpp:112]:
```suggestion
    cur += snprintf(cur, static_cast<uint16_t>(end - cur), "[(len:%d) ", aLength);
```

File: src/posix/platform/logging.cpp
Comment (bukepo) [reply src/posix/platform/logging.cpp:38]: The platform should only define otPlatLog when `LOG_OUTPUT == PLATFORM_DEFINED`. @abtink may I know how to reproduce the compile error you encountered?

File: src/posix/platform/trel_udp6.cpp
Comment (abtink) [reply src/posix/platform/trel_udp6.cpp:112]: This is really a c implementation not cpp file and therefore all other casts follow C style.
(we need to change the fule name to make it clear later in new PR).

File: src/posix/platform/logging.cpp
Comment (abtink) [reply src/posix/platform/logging.cpp]: This PR is intended to add back to our github action CI some TREL posix builds (somewhat addressing the issue https://github.com/openthread/openthread/issues/6472 caused due to removal of `ot-ncp` and previous TREL posix build checks and tests). 

The use of `LOG_OUTPUT_APP` with CLI under posix seems to have been broken (I guess we did not detect it because  this configuration was probably not being built/checked before). This config used to work correctly with NCP posix build (but as you know the `ot-ncp` was removed, so now we want to use CLI based builds).

I want this PR to be merged to somewhat address the lack any of build check for posix TREL build config. It does not make sense to deal with issues related to posix (I think they should be addressed in other PRs). The goal of this PR is to unblock other PRs adding changes in TREL like https://github.com/openthread/openthread/pull/6515 (so that at minimum we have a CI build check for TREL posix).

So in the new pushed commit in this PR I changed the `toranj` config to use `PLATFORM_DEFINED` for the posix builds, so we don't cover or use this config.

Note that use of `LOG_OUTPUT_APP` config is still broken on posix build. @bukepo , to reproduce simply set the `LOG_OUTPUT` to `LOG_OUTPUT_APP` and build posix CLI executable. 
- The reason for this seems to be that in the CLI, the `otPlatLog` is defined `examples/apps/cli/main.c` but in `posix` build does not seem to include this file (`posix` has its own `main.c` and the only place under posix where `otPlatLog` is defined is in `src/posix/platform/logging.cpp`).
- I leave it to you guys to address this (feel free to later change back the `toranj` config to use `OUTPUT_APP` under posix to add build coverage for this config).

Comment (bukepo) [reply src/posix/platform/logging.cpp]: Thanks for the explanation. The `LOG_OUTPUT_APP` was intentionally not implemented in posix ot-cli because we can get the same effect with `-v` option(see https://github.com/openthread/openthread/blob/main/src/posix/main.c#L367). The simulation platform is for testing purpose, so we should support as many configurations as possible on that platform. And in my opinion, we may just support what we need on POSIX platform. Thoughts? @abtink

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by jwhui — APPROVED** [review]

File: src/posix/platform/logging.cpp
Comment (simonlingoogle) [reply src/posix/platform/logging.cpp]: Can we define a weak `otPlatLog` in posix platform when `LOG_OUTPUT=LOG_OUTPUT_APP` ?

File: src/posix/platform/trel_udp6.cpp
Comment (simonlingoogle) [reply src/posix/platform/trel_udp6.cpp:112]: > This is really a c implementation not cpp file and therefore all other casts follow C style.

Is all Posix platform sources CPP? Is it beneficial to keep the code pure C?

Comment (abtink) [reply src/posix/platform/trel_udp6.cpp:112]: It is matter of style in each file. As I said, all other casts in this file are c format. l I don't have any preference here, but I don't think we want to change something like this from this PR.

**Pull request merged**
